### ISSUE TITLE
standard naming conventions

### ISSUE THREAD

**Issue by havok2063** [sdss/sdsstools#2]
Related to #1 , should we have a standard for the name of the config file?  Currently it's set to the `NAME` in the ``__init__`` at `~/.config/{NAME}/{NAME}.yml`.   Is this the name of the python package, i.e. the python directory name, or for example, the pip-package name?  I think there some inconsistencies with the `config` name, the `logger` name, and the `package` name for getting the version.   Should we allow these names to be flexible or set some standards amongst them? 

I think for most cases this is fine, but what about packages where the `pip` name and `package` name are different?  For example for `tree`, which has pip name `sdss-tree`, I've had to do this to get the config, version and log names to work out properly.  

```
from sdsstools import get_config, get_logger, get_package_version

NAME = 'sdss_tree'

# init the logger
log = get_logger(NAME)

# Loads config
config = get_config('tree')

__version__ = get_package_version(path=__file__, package_name=NAME)
```
Ideally I think I'd want the `logger` name to be `tree` as well, but only if a default log filepath was in some global `sdss` log directory.

**Comment by albireox** [comment]
In principle (but I may be wrong), `NAME` is always set to the package name (i.e., the pip name) so the `__version__` should work fine out of the box, at least with v2 of the python template.

Similarly, for the logger the name of the logger is set to `NAME`, i.e., the pip name. Note that this is just the name of the logger, not the file. To log to a file you need to call `log.start_file_logger()` which requires passing the name of the file you want.

For the configuration we are right now using the package name (in tree that would be `tree`, not `sdss-tree`) so there is some inconsistency there but since we are now storing the configuration files in `~/.config/sdss` I'd say that's fine because all config files are already under the umbrella of the sdss directory.

**Comment by havok2063** [comment]
Yeah that's what I figured.   It did but I had originally changed the `NAME` from `sdss-tree` to `tree` to match the package directory name and config file name and then was curious why `__version__` stopped working.  Ok, I didn't know that about the logger filename and `NAME` being different.  

I think I'm ok leaving things as is, just wanted to clarify.  I've added some additional comments for context in the `python-template`.
